cmd/influxd/run.go

```diff
@@ -71,6 +72,16 @@ func execRun(args []string) {
 		}
 		log.Printf("data node #%d listening on %s", s.ID(), config.DataAddr())
 
+		// Spin up the collectd server
+		if config.Collectd.Enabled {
+			c := config.Collectd
+			cs := collectd.NewServer(s, c.TypesDB)
+			cs.Database = c.Database
+			err := collectd.ListenAndServe(cs, c.ConnectionString(config.BindAddress))
```

Comment:
I believe @davecheney 's point about our API (it applies to the TCP and UDP Graphite servers too, which I wrote, the reason we have this pattern is primarily mine) is that it is not great. He thinks a better API is of this form:

`cs, err := collectd.ListenAndServer(add)`

He may be right. This would very much follow the pattern in the Go's net package. @davecheney ?